src/momento/auth/access_control/disposable_token_scope.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+from momento.auth.access_control.permission_scope import (
+    CachePermission,
+    CacheRole,
+    CacheSelector,
+    Permissions,
+)
+
+
+class AllCacheItems:
+    pass
+
+
+@dataclass
+class CacheItemKey:
+    key: str
+
+
+@dataclass
+class CacheItemKeyPrefix:
+    key_prefix: str
+
+
+@dataclass
+class CacheItemSelector:
+    cache_item: Union[CacheItemKey, CacheItemKeyPrefix, AllCacheItems, str]
+
+    def __init__(self, cache_item: Union[CacheItemKey, CacheItemKeyPrefix, AllCacheItems, str]):
+        self.cache_item = cache_item
```

Comment:
Tip: the dataclass decorator generates the init method based on the class variables you put above so this isn't necessary in any of the dataclasses (here and elsewhere). It generates other things as well such as `__eq__`, `__repr__`, `__hash__`, etc.